controllers/storagecluster/storageconsumer.go

```diff
@@ -104,8 +105,29 @@ func (s *storageConsumer) ensureCreated(r *StorageClusterReconciler, storageClus
 	return ctrl.Result{}, nil
 }
 
-func (s *storageConsumer) ensureDeleted(_ *StorageClusterReconciler, _ *ocsv1.StorageCluster) (ctrl.Result, error) {
-	// cleaned up via owner references
+func (s *storageConsumer) ensureDeleted(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (ctrl.Result, error) {
+	storageConsumer := &ocsv1a1.StorageConsumer{}
+	storageConsumer.Name = defaults.LocalStorageConsumerName
+	storageConsumer.Namespace = storageCluster.Namespace
+	if err := r.Client.Get(r.ctx, client.ObjectKeyFromObject(storageConsumer), storageConsumer); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get storageconsumer %s: %v", storageConsumer.Name, err)
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if controllerutil.RemoveFinalizer(storageConsumer, storageConsumerFinalizer) {
+		r.Log.Info("removing finalizer from StorageConsumer.", "StorageConsumer", storageConsumer.Name)
+		if err := r.Update(r.ctx, storageConsumer); err != nil {
+			r.Log.Info("Failed to remove finalizer from StorageConsumer", "StorageConsumer", storageConsumer.Name)
+			return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from StorageConsumer: %v", err)
+		}
+	}
+
+	if err := r.Client.Delete(r.ctx, storageConsumer); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to delete storageconsumer %s: %v", storageConsumer.Name, err)
+	}
```

Comment:
be consistent on how you are using embedded struct, `r.Update/r.Delete/r.Get` vs `r.Client.Update/r.Client.Delete/r.Client.Get`